var/spack/repos/builtin/packages/py-pyside2/package.py

```diff
@@ -30,13 +31,15 @@ class PyPyside2(PythonPackage):
         description="Enables the generation of html and man page documentation",
     )
 
-    depends_on("python@2.7.0:2.7,3.5.0:3.5,3.6.1:", type=("build", "run"))
-    depends_on("python@2.7.0:2.7,3.5.0:3.5,3.6.1:3.8", when="@:5.14", type=("build", "run"))
+    # dependency matrix from: https://wiki.qt.io/Qt_for_Python
+    depends_on("python@:3.10", when="@5.15.8:", type=("build", "run"))
+    depends_on("python@:3.9", when="@5.15.1:5.15.7", type=("build", "run"))
+    depends_on("python@:3.8", when="@:5.15.0", type=("build", "run"))
```

Comment:
I would add:
```python
depends_on("python@:3.7", when="@:5.13", type=("build", "run"))
```
Based on PyPI wheel availability. We'll be removing 3.7 in the next release (November) so it would be good to also remove those old versions of pyside2